Develop

martinyde opened the pull request.

Commit:
Added comment to class

Commit:
Merge branch 'master' into develop

Commit:
Added composer files

cableman commented on src/CprSniffer.php:
```diff
@@ -0,0 +1,183 @@
+     * @param $number
+     *   A string that could be a CPR number
+     *
+     * @return bool return True if it's considered a date
```
```suggestion
     * @return bool return 
     *   True if it's considered a date
```

cableman commented on src/CprSniffer.php:
```diff
@@ -0,0 +1,183 @@
+     *
+     * @return bool return True if it's considered a date
+     */
+    private function dateChk($number)
```
```suggestion
    private function dateChk(string $number): bool
```

cableman commented on src/CprSniffer.php:
```diff
@@ -0,0 +1,183 @@
+    /**
+     * Check against valid date.
+     *
+     * @param $number
```
```suggestion
     * @param string $number
```

cableman commented on src/CprSniffer.php:
```diff
@@ -0,0 +1,183 @@
+    /**
+     * Check against modulo 11.
+     *
+     * @param $array
```
```suggestion
     * @param array $array
```

cableman commented on src/CprSniffer.php:
```diff
@@ -0,0 +1,183 @@
+     *
+     * @param $array
+     *   An array of characters
+     * @param $number
```
```suggestion
     * @param string $number
```

cableman commented on src/CprSniffer.php:
```diff
@@ -0,0 +1,183 @@
+     * @param $number
+     *   A string that could be a CPR number
+     *
+     * @return bool True if the characters in union resemble a CPR number
```
```suggestion
     * @return bool 
     *   True if the characters in union resemble a CPR number
```

cableman commented on src/CprSniffer.php:
```diff
@@ -0,0 +1,183 @@
+     *
+     * @return bool True if the characters in union resemble a CPR number
+     */
+    private function mod11Chk($array, $number)
```
```suggestion
    private function mod11Chk(array $array, string $number): bool
```

cableman commented on src/CprSniffer.php:
```diff
@@ -0,0 +1,183 @@
+     *
+     * @return array the created array of characters
+     */
+    private function stringSplit($numString)
```
```suggestion
    private function stringSplit(string $numString): array
```

cableman commented on src/CprSniffer.php:
```diff
@@ -0,0 +1,183 @@
+     * @param $numString
+     *   A string to create an array from
+     *
+     * @return array the created array of characters
```
```suggestion
     * @return array<string>
     *   The created array of characters
```

cableman commented on src/CprSniffer.php:
```diff
@@ -0,0 +1,183 @@
+    }
+
+    /**
+     * Create an array from a string.
```
Why not use https://www.php.net/manual/en/function.str-split.php

> Converts a string to an array.

martinyde replied:
This was changed, and the function removed

cableman commented on src/CprSniffer.php:
```diff
@@ -0,0 +1,183 @@
+    /**
+     * Create an array from a string.
+     *
+     * @param $numString
```
```suggestion
     * @param string $numString
```

cableman commented on src/CprSniffer.php:
```diff
@@ -0,0 +1,183 @@
+     * @param $string
+     *   The string to check
+     *
+     * @return bool true if string contains a number that could be considered cpr
```
```suggestion
     * @return bool 
     *   True if string contains a number that could be considered cpr
```

cableman commented on src/CprSniffer.php:
```diff
@@ -0,0 +1,183 @@
+    /**
+     * Check if string contains valid CPR.
+     *
+     * @param $string
```
```suggestion
     * @param string $string
```

cableman commented on src/CprSniffer.php:
```diff
@@ -0,0 +1,183 @@
+     *
+     * @return bool true if string contains a number that could be considered cpr
+     */
+    public function checkCpr($string)
```
```suggestion
    public function checkCpr(string $string): bool
```

cableman commented on src/CprSniffer.php:
```diff
@@ -0,0 +1,183 @@
+<?php
+
```
```suggestion

declare(strict_types=1);

```

cableman reviewed (changes requested):
* Minor changes to enabled better type checking. 
* Maybe it should have been named `CPRValidator` not sure I understand the `sniffer` part.
* Maybe add php version 7.3 to composer (as php 7.2 and lower is EOL).

martinyde committed:
Update src/CprSniffer.php

Co-authored-by: Jesper Kristensen <jeskr@aarhus.dk>

martinyde committed:
Update src/CprSniffer.php

Co-authored-by: Jesper Kristensen <jeskr@aarhus.dk>

martinyde committed:
Update src/CprSniffer.php

Co-authored-by: Jesper Kristensen <jeskr@aarhus.dk>

martinyde committed:
Update src/CprSniffer.php

Co-authored-by: Jesper Kristensen <jeskr@aarhus.dk>

martinyde committed:
Update src/CprSniffer.php

Co-authored-by: Jesper Kristensen <jeskr@aarhus.dk>

martinyde committed:
Update src/CprSniffer.php

Co-authored-by: Jesper Kristensen <jeskr@aarhus.dk>

martinyde committed:
Update src/CprSniffer.php

Co-authored-by: Jesper Kristensen <jeskr@aarhus.dk>

martinyde committed:
Update src/CprSniffer.php

Co-authored-by: Jesper Kristensen <jeskr@aarhus.dk>

martinyde committed:
Update src/CprSniffer.php

Co-authored-by: Jesper Kristensen <jeskr@aarhus.dk>

martinyde committed:
Update src/CprSniffer.php

Co-authored-by: Jesper Kristensen <jeskr@aarhus.dk>

martinyde committed:
Update src/CprSniffer.php

Co-authored-by: Jesper Kristensen <jeskr@aarhus.dk>

martinyde committed:
Update src/CprSniffer.php

Co-authored-by: Jesper Kristensen <jeskr@aarhus.dk>

martinyde committed:
Update src/CprSniffer.php

Co-authored-by: Jesper Kristensen <jeskr@aarhus.dk>

martinyde committed:
Update src/CprSniffer.php

Co-authored-by: Jesper Kristensen <jeskr@aarhus.dk>

Commit:
Added codesniffer and changed string split

Commit:
Added php dependency

martinyde commented:
>     * Minor changes to enabled better type checking.
> 
>     * Maybe it should have been named `CPRValidator` not sure I understand the `sniffer` part.
> 
>     * Maybe add php version 7.3 to composer (as php 7.2 and lower is EOL).

- Added your suggested changes
- Added codesniffer config file.
- One lined variables and return value commenting in accordance with codesniffer.
- Added php 7.3 dependency
- About naming i grabbed the sniffer part from the name of the jira issue, thought it was your own naming. I'm not really invested in what the name should be so fell free to change.
